gwcs/wcs.py

```diff
@@ -737,80 +748,53 @@ def to_fits_sip(self, bounding_box, max_error=None, max_rms=None, degree=None,
         ipdegree = fit_inv_poly_u.degree
         hdr['ap_order'] = ipdegree
         hdr['bp_order'] = ipdegree
-        cd, sip_poly_x, sip_poly_y = reform_poly_coefficients(fit_poly_x, fit_poly_y)
-        store_2D_coefficients(hdr, sip_poly_x, 'A')
-        store_2D_coefficients(hdr, sip_poly_y, 'B')
-        store_2D_coefficients(hdr, fit_inv_poly_u, 'AP', keeplinear=True)
-        store_2D_coefficients(hdr, fit_inv_poly_v, 'BP', keeplinear=True)
+        cd, sip_poly_x, sip_poly_y = _reform_poly_coefficients(fit_poly_x, fit_poly_y)
+        _store_2D_coefficients(hdr, sip_poly_x, 'A')
+        _store_2D_coefficients(hdr, sip_poly_y, 'B')
+        _store_2D_coefficients(hdr, fit_inv_poly_u, 'AP', keeplinear=True)
+        _store_2D_coefficients(hdr, fit_inv_poly_v, 'BP', keeplinear=True)
         hdr['cd1_1'] = cd[0][0]
         hdr['cd1_2'] = cd[0][1]
         hdr['cd2_1'] = cd[1][0]
         hdr['cd2_2'] = cd[1][1]
         hdr['sipmxerr'] = max_resid
         hdr['siprms'] = rms
-        # # for test purposes, make sip equivalent in GWCS
-        # cdmat = np.array([[cd[0][0], cd[0][1]], [cd[1][0], cd[1][1]]])
-        # # cdmat = [[cd[0][0], cd[1][0]], [cd[0][1], cd[1][1]]]
-        # sip_poly_test_x = sip_poly_x.copy()
-        # sip_poly_test_y = sip_poly_y.copy()
-        # # sip_poly_test_x.c1_0 = 1
-        # # sip_poly_test_y.c0_1 = 1
-        # shift = (Shift(-crpix1) & Shift(-crpix2))
-        # aff = AffineTransformation2D(cdmat, [0,0])
-
-        # transform = (shift | Mapping([0,1,0,1]) |
-        #             (Mapping([0], n_inputs=2) + sip_poly_test_x) &
-        #             (Mapping([1], n_inputs=2) + sip_poly_test_y) |
-        #             aff |
-        #             Pix2Sky_TAN() | RotateNative2Celestial(crval1, crval2, 180))
-        # transform2 = ((Shift(-crpix1) & Shift(-crpix2)) | Mapping([0,1,0,1]) |
-        #               (fit_poly_x & fit_poly_y) | Pix2Sky_TAN() |
-        #               RotateNative2Celestial(crval1, crval2, 180))
         return hdr
 
 
-def fit_2D_poly(degree, max_error, max_rms, u, v, x, y, verbose=False):
+def _fit_2D_poly(degree, max_error, max_rms, u, v, x, y, verbose=False):
     """
     Fit a pair of ordinary 2D polynomial to the supplied inputs (u, v) and
     outputs (x, y)
     """
     llsqfitter = LinearLSQFitter()
     # The case of one pass with the specified polynomial degree
-    if degree is not None:
-        poly_x = Polynomial2D(degree=degree)
-        poly_y = Polynomial2D(degree=degree)
+    if degree is None:
+        deglist = range(2,10)
```

Comment:
I really think that `range` should start at 1 and then if that is enough to meet user's requirement for accuracy, just do not write out SIP coefficients.

Reply:
It is a bit more involved in that since it also means changing the WCS type, but sure I can add that.